storage/storage_image.go

```diff
@@ -941,7 +941,7 @@ func (s *storageImageDestination) commitLayer(ctx context.Context, blob manifest
 	s.lock.Unlock()
 	if ok {
 		layer, err := al.PutAs(id, lastLayer, nil)
-		if err != nil {
+		if err != nil && errors.Cause(err) != storage.ErrDuplicateID {
```

Comment:
Would using the new standard-library `errors.Is` work?

`errors.Cause` from `pkg/errors` is a superset of `errors.Is`, so it’s not a trivial substitution. And because `pkg/errors` is frozen`, it would be nice not to add any more users.